server/instance.go

```diff
@@ -125,8 +128,24 @@ func (s *Server) registerInstanceRoutes(g *echo.Group) {
 		if err := jsonapi.UnmarshalPayload(c.Request().Body, instancePatch); err != nil {
 			return echo.NewHTTPError(http.StatusBadRequest, "Malformatted patch instance request").SetInternal(err)
 		}
+		instance, err := s.composeInstanceByID(ctx, id)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to get instance ID: %v", id)).SetInternal(err)
+		}
+		if instance == nil {
+			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("Instance ID not found: %d", id))
+		}
+		host, port := instance.Host, instance.Port
+		if instancePatch.Host != nil {
+			host = *instancePatch.Host
+		}
+		if instancePatch.Port != nil {
+			port = *instancePatch.Port
+		}
```

Comment:
Do we still need this?

Reply:
Yeah, we need to validate the final host/port values to be updated to.